lib/Sema/TypeCheckDecl.cpp

```diff
@@ -680,8 +680,12 @@ ExistentialRequiresAnyRequest::evaluate(Evaluator &evaluator,
 
     // For value members, look at their type signatures.
     if (auto valueMember = dyn_cast<ValueDecl>(member)) {
-      if (!decl->isAvailableInExistential(valueMember))
+      const auto info = valueMember->findProtocolSelfReferences(
+          decl, /*treatNonResultCovariantSelfAsInvariant=*/false);
+      if (info.selfRef > SelfReferencePosition::Covariant ||
+          info.assocTypeRef) {
         return true;
+      }
```

Comment:
@hborla Does this match your interpretation of a non-covariant `Self` reference?

Reply:
I believe so. For some context, this request is used for a syntax restriction for pre-Swift 6 code. The reason is so that folks don't start to write new existential code using SE-0309 that will become invalid in Swift 6 by requiring `any`. It's better for the migration if they just write `any` out of the gate, and this also helps communicate the existence of the `any` keyword.

(obviously this is also pending the review decision for SE-0335, and the use of this request to produce an error message is behind a flag)